### ISSUE TITLE
Question about seed dynamic updating

### ISSUE THREAD

**Issue by DotWang** [speedinghzl/DSRG#22]
It's a very interesting work. We are conducting a similar segmentation task without initial cues file, and I hope to generate seeds in the first epoch and update them in subsequent epoches. 
However, in **pylayers.py** I think the grown seeds (through SRG method) that load from the outside pickel file are directly used to compute seed loss, meaning the cues pickle file needed to be imported at every epoch, which seems different from the paper saying that current seeds map are obtained through the last epoch generated seeds map.
I don't know when and where the previous seeds (not the loaded pickle file) are substituted with the updated seeds and used as the newest seeds, can you help me? 
Thank you.

**Comment by speedinghzl** [comment]
Thanks for your interest in our work. The seed cues which stored in the pickle file is very important. In every training iteration, the current supervision grows starting from the original seed cues rather than the last iteration generated seeds. And you can find the sentence "To ensure the stability of training, DSRG always chooses the original seed cues as initial seed points" in our introduction section. In a word, we do not perform the seeded region growing with the previous seeds.

**Comment by DotWang** [comment]
Thanks for your clear explaination. So the bottom row results in Figure 1 whether can be understood as for a certain input image, in every epoch, although a same cues pickle file is always loaded as the initial seed , the label map can still dynamic updating because of the constantly changed CAM during training?

**Comment by speedinghzl** [comment]
The label map can still dynamic update because of the constantly changed segmentation networks during training. We use the CAM to generate the initial seed (cues pickle file) offline. Afterward, we have not used it.

**Comment by DotWang** [comment]
OK, I see. thank you very much. (It's my fault, the "CAM" I just said is the probability map "fc8-SEC-Softmax" in the code)
